clusterloader2/pkg/measurement/common/simple/profile.go

```diff
@@ -201,22 +202,32 @@ func (*cpuProfileMeasurement) String() string {
 	return cpuProfileName
 }
 
-func (p *profileMeasurement) gatherProfile() (measurement.Summary, error) {
+func (p *profileMeasurement) gatherProfile(c clientset.Interface) (measurement.Summary, error) {
 	profilePort, err := getPortForComponent(p.config.componentName)
 	if err != nil {
 		return nil, fmt.Errorf("profile gathering failed finding component port: %v", err)
 	}
 
-	// Get the profile data over SSH.
-	getCommand := fmt.Sprintf("curl -s localhost:%v/debug/pprof/%s", profilePort, p.config.kind)
-	sshResult, err := measurementutil.SSH(getCommand, p.config.host+":22", p.config.provider)
-	if err != nil {
-		if p.config.provider == "gke" {
-			// Only logging error for gke. SSHing to gke master is not supported.
-			klog.Errorf("%s: failed to execute curl command on master through SSH: %v", p.name, err)
-			return nil, nil
+	var profile string
```

Comment:
The above won't work - getPortForComponent will return an error for kube-apiserver.
You need to handle kube-apiserver first, and only the getPortForComponent and the rest (btw you also won't have "else" which is always good :)

Reply:
Done.
For some reason I was positive that I moved getPortForComponent to else section...